config/routes.rb

```diff
@@ -0,0 +1,10 @@
+Rails.application.routes.draw do
+  get 'user/:user_id/posts' => 'posts#index', as: 'user_posts'
+  get 'users/:user_id/posts/:id' => 'posts#show', as: 'user_post'
+  get 'users' => 'users#index', as: 'users'
+  get 'users/:id' => 'users#show', as: 'user'
```

Comment:
- [ ] <b>REQUIRED:</b> Here is more preferred way of setting up routes, Kindly do that to comply with project requirements.
```rb
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```